activerecord/lib/active_record/scoping/named.rb

```diff
@@ -30,13 +30,8 @@ def all
         end
 
         def default_scoped # :nodoc:
-          scope = build_default_scope
-
-          if scope
-            relation.spawn.merge!(scope)
-          else
-            relation
-          end
+          scope = relation
+          build_default_scope(scope) || scope
```

Comment:
I refactored `default_scoped` to avoid creating extra relation and merging.
it will make `klass.all` a little faster.

```
CollectionProxies (1000000):   1.420000   0.010000   1.430000 (  1.517390)
Associations (1000000):   0.120000   0.000000   0.120000 (  0.121722)
```